space_inv/main.py

```diff
@@ -41,29 +40,18 @@ def spawner(self):
             self.alien_spawner = 50
 
     def spawn_alien(self):
-        x = choice([randint(0, WIDTH), choice([-self.border, WIDTH + self.border])])
-        if (x == -self.border) or (x == WIDTH + self.border):
-            y = randint(-self.border, HEIGHT + self.border)
-        else:
-            y = choice([-self.border, HEIGHT + self.border])
-        alien_sprite = Alien(choice(["red", "yellow", "green"]), x, y)
+        alien_sprite = Alien.create()
         self.aliens.add(alien_sprite)
 
     def kill_alien(self):
-        to_add = len(self.aliens)
         for laser in self.player.sprite.lasers:
-            if pygame.sprite.spritecollide(laser, self.aliens, True):
-                self.coins += to_add - len(self.aliens)
+            aliens_hit = pygame.sprite.spritecollide(laser, self.aliens, True)

```

Comment:
I don't know how you want to solve the mechanics of killing aliens, but currently, since aliens don't have collisions, it is possible to kill many of them with one laser. This could be changed to some kind of laser penetration mechanics

Reply:
Yeah I've changed that the last time I was doing this project, the version I told you about but haven't uploaded it here